### PR TITLE
[Importer] Add logic for wrapping negative axes.

### PR DESCRIPTION
**Pull request by mciprian13**
**Summary**
Add logic for wrapping negative axes for some operators (e.g. Softmax, Slice, Concat, Squeeze, etc) since the proto format allows it but Glow does not handle properly negative axes.

**Fixes**
#4004

**Test Plan**
Current tests are passing.
Added unit test for the utility function.

### CODE REVIEW

File: lib/Importer/ProtobufLoader.cpp [hunk 1]
```diff
@@ -35,6 +35,16 @@ bool isArrayConstant(llvm::ArrayRef<size_t> a) {
...
 }
 
+Expected<int> getPositiveAxis(int axis, int rank) {
+  RETURN_ERR_IF_NOT(-rank <= axis, "Axis parameter invalid!");
+  RETURN_ERR_IF_NOT(axis < rank, "Axis parameter invalid!");
+  int axisPos = axis % rank;
+  if (axisPos < 0) {
+    axisPos += rank;
+  }
+  return axisPos;
```
Comment (jackm321) [inline lib/Importer/ProtobufLoader.cpp:45]: Give the above preconditions `-rank <= axis < rank` why is % needed?

File: lib/Importer/ProtobufLoader.cpp [hunk 2]
```diff
@@ -35,6 +35,16 @@ bool isArrayConstant(llvm::ArrayRef<size_t> a) {
   return true;
 }
 
+Expected<int> getPositiveAxis(int axis, int rank) {
+  RETURN_ERR_IF_NOT(-rank <= axis, "Axis parameter invalid!");
+  RETURN_ERR_IF_NOT(axis < rank, "Axis parameter invalid!");
```
Comment (jackm321) [inline lib/Importer/ProtobufLoader.cpp:40]: Can you add axis and rank to the messages with strFormat

File: include/glow/Importer/ProtobufLoader.h [hunk 3]
```diff
@@ -131,6 +131,40 @@ template <typename T> std::string loadOperatorName(const T &op) {
...
+}
+
+/// Reads multiple axes as an array which are wrapped if negative using \p rank
+/// based on the logic of \ref getPositiveAxis.
+template <typename ElemTy, typename T> static
+Expected<std::vector<ElemTy>> loadAxes(const T *arg, int rank) {
+  std::vector<int> axes;
+  ASSIGN_VALUE_OR_RETURN_ERR(axes, getShape<int>(arg));
+  std::vector<ElemTy> axesPos;
+  for (const int &axis: axes) {
```
Comment (jackm321) [inline include/glow/Importer/ProtobufLoader.h]: super nit, no reason to have ref to int

File: lib/Importer/ProtobufLoader.cpp (hunk 1 above)
Comment (mciprian13) [reply lib/Importer/ProtobufLoader.cpp:45]: Yeah you`re right ... I actually designed the logic in the first place without those verifications. I`ll change it. Thanks!

Comment (mciprian13) [reply lib/Importer/ProtobufLoader.cpp]: Done.

File: lib/Importer/ProtobufLoader.cpp (hunk 2 above)
Comment (mciprian13) [reply lib/Importer/ProtobufLoader.cpp]: Done.

File: include/glow/Importer/ProtobufLoader.h (hunk 3 above)
Comment (mciprian13) [reply include/glow/Importer/ProtobufLoader.h]: Done.

**Comment by mciprian13** [comment]
@jackm321 Regarding the ONNX test coverage: I once proposed plugging Glow in the ONNX automated test framework which has a huge set of unit tests for all the ONNX operators and provides a report for ONNX compliance (details here #3809). Designing sample models for testing the ONNX importer is time consuming ... I think this is the way to go.
I read some details of doing this integration and requires some skills about containers (which I don't have) but I could help with something else maybe ... Don't you think this provides value for Glow?

**Comment by jackm321** [comment]
The pytorch test is flakey, will look into that today

**Comment by mciprian13** [comment]
@jackm321 Thanks for pulling this so quick!
